extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -9,203 +12,306 @@ class SpecialDiscussionsLogController extends WikiaSpecialPageController {
-				$userLogRecord->language = $record->{'mobile_app.event.device_language'};
-				$userLogRecord->location =
-					$record->{'mobile_app.geo_ip.city'} . ', ' . $record->{'mobile_app.geo_ip.country_name'};
-				$userLogRecord->timestamp = date( DATE_RFC2822, $rawTimestamp / 1000 );
-				$userLogRecord->userAgent =
-					$record->{'mobile_app.data.platform'} . ' ' . $record->{'mobile_app.data.platform_version'};
-				$userLogRecord->userId = $userId;
-				$userLogRecord->userName = $userName;
-				$records[ $rawTimestamp ] = $userLogRecord;
-			}
+			$this->appendRecordsFromResponse( $records, $response, $ipHash );
 		}
 
 		krsort( $records, SORT_NUMERIC );
 		return $records;
 	}
 
-	private function getQuery( $userId ) {
-		$pagination_size = self::PAGINATION_SIZE;
-		return <<<JSON_BODY
-{
-	"query": {
-		"filtered": {
-			"query": {
-				"bool": {
-					"should": [{
-						"query_string": {
-							"query":"mobile_app.event.user_id:$userId"
-						}
-					}]
-				}
-			}
+	private function performSearchRequest( $url, $query ) {
+		$client = new \GuzzleHttp\Client();
+
+		try {
+			$response = $client->post(
+				$url, [
+					'body' => $query,
+				]
+			);
+		} catch ( \GuzzleHttp\Exception\RequestException $requestException ) {
+			$this->logger->error(
+				sprintf( 'Request to elasticsearch failed: %s', $requestException->getMessage() ),
+				[ 'exception' => $requestException ]
+			);
+			return false;
 		}
-	},
-	"size":$pagination_size,
-	"sort":[{
-		"@timestamp": {
-			"order": "desc"
+
+		if ( $response->getStatusCode() !== self::HTTP_STATUS_OK ) {
+			$this->logger->error(
+				sprintf( 'Elasticsearch request error; status code %d', $response->getStatusCode() )
+			);
+			return false;
 		}
-	}]
-}
-JSON_BODY;
+
+		return $response;
 	}
 
-	private function checkAccess() {
-		if ( !$this->wg->User->isAllowed( 'specialdiscussionslog' ) ) {
-			return false;
+	private function appendRecordsFromResponse( array &$records, $response, array &$ipHash ) {
+
+		$resultObject = json_decode( $response->getBody() );
+		$hits = $resultObject->hits->hits;
+
+		foreach ( $hits as $hit ) {
+			$record = $hit->_source;
+			$ip = $record->{'mobile_app.client_ip'};
+			$app = $record->{'mobile_app.data.app_name'} . ' ' . $record->{'mobile_app.data.app_version'};
+			$ipHashKey = $ip . ':' . $app;
+
+			// Filter out records with duplicate ip/app
+			if ( !empty( $ipHash[$ipHashKey] ) ) {
+				continue;
+			}
+
+			try {
+				$user = $this->getUserById( $record->{'mobile_app.event.user_id'} );
+			} catch ( \Exception $e ) {
+				$this->logger->error(
+					sprintf( 'User not found: %s', $e->getMessage() ),
+					[ 'exception' => $e ]
+				);
+				continue;
+			}
+			$rawTimestamp = $record->{'mobile_app.event.timestamp'};
+			$userLogRecord = new UserLogRecord();
+			$userLogRecord->app = $app;
+			$userLogRecord->ip = $ip;
+			$userLogRecord->language = $record->{'mobile_app.event.device_language'};
+			$city = $record->{'mobile_app.geo_ip.city'};
```

Comment:
How about we get rid of this local variable and just use `$record->{'mobile_app.geo_ip.city'}` for `$city` on the line below? For me the visual consistency of setting something to `$userLogRecord` on every line beats trying to break up a longer line with a local var.